prometheus/exporter/job_exporter.py

```diff
@@ -151,36 +144,45 @@ def wrapper(semaphore, queue):
         if now - self.old_metrics_time < self.old_data_timeout_s:
             return self.old_metrics
 
-        logger.info("result is too old")
+        logger.info("gpu info is too old")
         return None
 
 def main(argv):
     logDir = argv[0]
+    gpuMetricsPath = logDir + "/gpu_exporter.prom"
+    jobMetricsPath = logDir + "/job_exporter.prom"
+
     timeSleep = int(argv[1])
     iter = 0
 
-    gpu_metrics_getter = functools.partial(gpu_exporter.gen_gpu_metrics_from_smi, logDir)
-
-    singleton = Singleton(gpu_metrics_getter)
+    singleton = Singleton(gpu_exporter.collect_gpu_info)
 
     while True:
         try:
             logger.info("job exporter running {0} iteration".format(str(iter)))
             iter += 1
-            gpu_metrics = singleton.try_get()
+            gpuInfos = singleton.try_get()
+
+            gpuMetrics = gpu_exporter.convert_gpu_info_to_metrics(gpuInfos)
+            if gpuMetrics is not None:
+                utils.export_metrics_to_file(gpuMetricsPath, gpuMetrics)
+
             # join with docker stats metrics and docker inspect labels
-            gen_job_metrics(logDir, gpu_metrics)
+            jobMetrics = collect_job_metrics(gpuInfos)
+            utils.export_metrics_to_file(jobMetricsPath, jobMetrics)
```

Comment:
not seen this utils code at master branch

Reply:
ah, I forget to add it.